release/src/main/groovy/run_release_candidate_python_validation.sh

```diff
@@ -0,0 +1,298 @@
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+#  This file will verify Apache/Beam release candidate python by following steps:
+#
+#  1. Download files from RC staging location
+#  2. Verify hashes
+#  3. Create a new virtualenv and install the SDK
+#  4. Run Wordcount examples with DirectRunner
+#  5. Run Wordcount examples with DataflowRunner
+#  6. Run streaming wordcount on DirectRunner
+#  7. Run streaming wordcount on DataflowRunner
+#
+
+set -e
+set -v
+
+print_separator() {
+    echo "############################################################################"
+    echo $1
+    echo "############################################################################"
+}
+
+function get_version {
+    # this function will pull python sdk version from sdk/python/apache_beam/version.py and eliminate postfix '.dev'
+    version=$(awk '/__version__/{print $3}' sdks/python/apache_beam/version.py)
+    echo $version | cut -c 2- | rev | cut -d'.' -f2- | rev
+}
+
+update_gcloud() {
+    curl https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-189.0.0-linux-x86_64.tar.gz \
+    --output gcloud.tar.gz
+    tar xf gcloud.tar.gz
+    ./google-cloud-sdk/install.sh --quiet
+    . ./google-cloud-sdk/path.bash.inc
+    gcloud components update --quiet || echo 'gcloud components update failed'
+    gcloud -v
+}
+
+complete() {
+    print_separator "Validation $1"
+    rm -rf $TMPDIR
+}
+
+run_pubsub_publish(){
+    words=("hello world!", "I like cats!", "Python", "hello Python", "hello Python")
+    for word in ${words[@]}
+    do
+        gcloud pubsub topics publish $PUBSUB_TOPIC1 --message "$word"
+    done
+    sleep 10
+}
+
+run_pubsub_pull(){
+    gcloud pubsub subscriptions pull --project=$PROJECT_ID $PUBSUB_SUBSCRIPTION --limit=100 --auto-ack
+}
+
+create_pubsub(){
+    gcloud pubsub topics create --project=$PROJECT_ID $PUBSUB_TOPIC1
+    gcloud pubsub topics create --project=$PROJECT_ID $PUBSUB_TOPIC2
+    gcloud pubsub subscriptions create --project=$PROJECT_ID $PUBSUB_SUBSCRIPTION --topic $PUBSUB_TOPIC2
+}
+
+cleanup_pubsub(){
+    gcloud pubsub topics delete --project=$PROJECT_ID $PUBSUB_TOPIC1
+    gcloud pubsub topics delete --project=$PROJECT_ID $PUBSUB_TOPIC2
+    gcloud pubsub subscriptions delete --project=$PROJECT_ID $PUBSUB_SUBSCRIPTION
+}
+
+
+# Python Release Candidate Configuration
+echo "SDK version: $(get_version)"
+VERSION="2.3.0"
```

Comment:
call `get_version` instead?

Reply:
Yes, will do. I currently hard coded 2.3.0 for test purpose. Since 2.4.0 release is not cut yet, use get_version will fail the test. I will replace the hard-coding in the end.